enhancements/validate-catalog-for-bad-bundles.md

```diff
@@ -0,0 +1,177 @@
+apiVersion: policy/v1beta1
+kind: PodDisruptionBudget
+metadata:
+  name: zk-pdb
+spec:
+  minAvailable: 2  
+  selector:
+    matchLabels:
+      app: zookeeper
+```
+- Either `minAvailable` or `maxUnavailable` should be included, but not both
+- `minAvailable` field cannot be set to 0 or less
+- `maxUnavailable` cannot be set to greater than the number of pods (TODO: check with Dan)
+
+
+### PriorityClass yaml validation
+Example PriorityClass.yaml
+```yaml
+apiVersion: scheduling.k8s.io/v1
+kind: PriorityClass
+metadata:
+  name: high-priority
+value: 1000000
+globalDefault: false
+description: "This priority class should be used for XYZ service pods only."
+```
+- `globalDefault` should always be set to false
+
+
+### VerticalPodAutoScaler yaml validation
+Example VerticalPodAutoScaler.yaml
+```yaml
+apiVersion: autoscaling.k8s.io/v1
+kind: VerticalPodAutoscaler
+metadata:
+  name: my-rec-vpa
+spec:
+  targetRef:
+    apiVersion: "apps/v1"
+    kind:       Deployment
+    name:       my-rec-deployment
+  updatePolicy:
+    updateMode: "Off"
+```
+- 
+
+## Proposal
+
+The design proposal has two parts:
+- The first discusses potential approaches for validating SAP Gardener objects.
+- The second part discusses design approach for having a tool to list "bad" bundles in a catalog
+
+## Part 1: Validate bundle objects
+
+### 1. Extending Validation library to add validations in `Golang`
+- Extend [BundleValidator](https://github.com/operator-framework/api/blob/master/pkg/validation/internal/bundle.go#L14) to include new validation functions for each of the SAP Gardener objects. (ToDo: Do we add any checks under [BundleFormat](https://github.com/operator-framework/operator-registry/blob/master/pkg/lib/bundle/validate.go#L65) ?)
+- The validation function for each kube manifest object would check for conditions specified under [Static Validation Rules](https://hackmd.io/7TlDq9t_RuGjGm8YsEbyHA?both#Static-Validation-Rules).
+- (ToDo: 
+    -    Do we want to maintain some sort of `struct` for storing various kube-manifest objects as these objects could grow (in the future) or
+    -    Wait to see how these new objects are represented in the [Bundle](https://github.com/operator-framework/api/blob/master/pkg/manifests/bundle.go#L11) as part SAP gardener PR and then take a call)
+ 
+### 2. Using [CUE](https://cuelang.org/docs/integrations/yaml/#validate-yaml-files)
+- CUE is a configuration language that allows you to define a `<pdb-checks>.cue` template for validating Kube Manifests like PDBs.
+- CUE can be an extendable validation library to accept validate template for a resource like PDBs before applying them on to a cluster.
+
+
+#### Pros
+- Not hinging on a third-party operator like OPA Gatekeeper.
+- Validation is extendable to customized .cue files
+
+#### Cons
+- Not a lot to offer in terms of flexibility as compared to [OpenPolicyAgent (OPA)](https://kubernetes.io/blog/2019/08/06/opa-gatekeeper-policy-and-governance-for-kubernetes/)
```

Comment:
Could you add more detail here? What are the advantages in terms of flexibility we would get from OPA that we wouldn't get from CUElang?

Reply:
I have updated the doc. With OPA, you can define custom policies for create, update and delete of Kubernetes objects and these can be validated on cluster. CUE is more of a pre-flight checklist (off cluster)